### ISSUE TITLE
Error on first attempt to `docker-compose up` 

### ISSUE THREAD

**Issue by ssteinerx** [mhelmer/dockerize-uwsgi-nginx-webpack#11]
Running OS X, Docker Version 1.12.3-beta30.1 (13946)  (latest)

Just pulled from master, and decided to jump straight in and see what comes up by default.

When building react, all goes fine 'till step 3 when this happens:

```
Step 3 : RUN useradd -s /bin/bash --uid 1000 --home-dir /srv/react react
 ---> Running in 2e5452d415a1
useradd: UID 1000 is not unique
ERROR: Service 'react' failed to build: The command '/bin/sh -c useradd -s /bin/bash 
--uid 1000 --home-dir /srv/react react' returned a non-zero code: 4
```

**Comment by ssteinerx** [comment]
Just to try to get running, I changed the --uid to 1010.  Attempting to follow the instructions under:

### Development

```
...output of docker-compose build...
Successfully built 22ea02407510
bash-3.2$ docker-compose run nginx chown -R 1000:1000 /srv/*
Starting dockerizeuwsginginxwebpack_django_1
chown: cannot access '/srv/*': No such file or directory
```

I stopped there as each thing I change seems to lead to another issue.

I'm guessing that there are, e.g. `.env` files in your develop environment that are making this work, but they are `.gitignored` and so are not available when checking out this repo.

**Comment by mhelmer** [comment]
I don't think this should have anything to do with my `.env` but I will make a fresh clone to check it out later. My initial thought is that something has change with the node image we are building from. Will do a rebuild on travis.

**Comment by mhelmer** [comment]
To confirm what I suspected:
- The build is now failing on what used to be a success commit: https://travis-ci.org/mhelmer/dockerize-uwsgi-nginx-webpack/builds/175457553
- Seems like they are already creating a node user with uid 1000. https://github.com/nodejs/docker-node/commit/4a722c29c0e52624af8b72b4711ebeba8ea39463

The reason I create a user with uid 1000 is to
1. Not run as root
2. Assuming that your local user has UID 1000, if you mount a volume, files created by docker and your local user will have the same owner.

**Comment by mhelmer** [comment]
I've updated the `Dockerfile` to use the pre-existing `node` user. It should be fixed now. As for your second problem, I would test to `docker-compose up` once before running that command.

**Comment by ssteinerx** [comment]
Built!  Took an abominably long time (hours).  Slow tasks were ch[mod|own] and node-sass (which may have just been because it was last and all the cleanup after a bunch of installs prevented return.

Then, two of the containers crash with missing dependencies.

[startup log](https://gist.github.com/ssteinerx/32d66c0632d69330d5fa996dfd3d7db7)

**Comment by mhelmer** [comment]
Hmm. It's usually pretty fast for me (as is the jenkins build), but I could do some optimizations with `find` so it would only chown files with root-owner.

I should probably explain a bit how it works and what it different in dev and prod build.

Basically, the `Dockerfiles` will build the full image, including dependencies, and place that in `/srv/{react,django}`. If you use the production compose-file, it will use those files.

However, when you are developing things you really don't want to re-build the container everytime you change something. To solve that problem, we set it up to mount `react/src` from your host at `/srv/react` in the react container, and `django/src` at `/srv/django` in the django container.

This mean that when you run in development, your containers won't see the things that were built into `/srv/{react,django}` since your local directories are mounted there. This includes `node_modules` and `.env`.

To populate them, you should run:

```bash
# New docker volumes are created with root as owner.
docker-compose run nginx chown -R 1000:1000 /srv/*

docker-compose run django ./bootstrap-dev # installs virtualenv in .env
docker-compose run react npm install
```

**Comment by mhelmer** [comment]
Closing. The original error was solved by 9f71794. Feel free to open a new issue if you have more problems.
